### ISSUE TITLE
SparkSQL Comment section not aligned in one go

### ISSUE THREAD

**Issue by Aigul9** [sqlfluff/sqlfluff#7510]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I mentioned it in the issue: https://github.com/sqlfluff/sqlfluff/issues/7428
But upgrading to 4.0.4 didn't help

### Expected Behaviour

Data types and comments should be aligned in one go

### Observed Behaviour

Initial script:
```
CREATE TABLE IF NOT EXISTS dic_date (
    date_pk   DATE           COMMENT 'Date',
    year_num                    INT    COMMENT 'Year',
```

First run:
```
CREATE TABLE IF NOT EXISTS dic_date (
    date_pk                     DATE                     COMMENT 'Date',
    year_num                    INT    COMMENT 'Year',
```

```
sqlfluff fix migrations/lakehouse/dic/tables/dic_date.sql -v -p 0
==== sqlfluff ====
sqlfluff:                4.0.4 python:                 3.14.3
implementation:        cpython verbosity:                   1
dialect:              sparksql templater:         placeholder
rules:                                all
==== finding fixable violations ====
== [migrations/lakehouse/dic/tables/dic_date.sql] FAIL
L:   4 | P:  12 | LT01 | 'data_type' elements are expected to be aligned. Found
                       | incorrect whitespace before 'DATE' keyword: '   '.
                       | [layout.spacing]
L:   4 | P:  19 | LT01 | 'column_constraint_segment' elements are expected to be
                       | aligned. Found incorrect whitespace before 'COMMENT'
                       | keyword: '           '. [layout.spacing]
== [migrations/lakehouse/dic/tables/dic_date.sql] FIXED
2 fixable linting violations found
```

Second run:
```
CREATE TABLE IF NOT EXISTS dic_date (
    date_pk                     DATE   COMMENT 'Date',
    year_num                    INT    COMMENT 'Year',
```

```
sqlfluff fix migrations/lakehouse/dic/tables/dic_date.sql -v -p 0
==== sqlfluff ====
sqlfluff:                4.0.4 python:                 3.14.3
implementation:        cpython verbosity:                   1
dialect:              sparksql templater:         placeholder
rules:                                all
==== finding fixable violations ====
== [migrations/lakehouse/dic/tables/dic_date.sql] FAIL
L:   4 | P:  37 | LT01 | 'column_constraint_segment' elements are expected to be
                       | aligned. Found incorrect whitespace before 'COMMENT'
                       | keyword: '                     '.
                       | [layout.spacing]
== [migrations/lakehouse/dic/tables/dic_date.sql] FIXED
1 fixable linting violations found
```

### How to reproduce

1. Create a table like this:
```
CREATE TABLE IF NOT EXISTS dic_date (
    date_pk                  DATE                     COMMENT 'Date',
    year_num                    INT    COMMENT 'Year'
)
USING ICEBERG;
```
2. Run sqlfluff

### Dialect

sparksql

### Version

4.0.4

### Configuration

-

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
You have NOT shared your configuration, so I cannot really help.

Out of the box it would not align. You need to set at least the following configuration in your `.sqlfluff` file:
```ini
[sqlfluff]
# use_rust_parser = True
dialect = sparksql

[sqlfluff:layout:type:data_type]
spacing_before = align
align_within = statement

[sqlfluff:layout:type:column_constraint_segment]
spacing_before = align
align_within = statement%                                   
```

```bash
cat test.sql
CREATE TABLE mytable (
    date_pk               STRING        COMMENT 'Date',
    year_num        INT  COMMENT 'Year'
)
```

```bash
sqlfluff fix test.sql -v -p 0
==== sqlfluff ====
sqlfluff:          4.0.4   python:                 3.12.3
implementation:        cpython verbosity:                   1
dialect:              sparksql templater:               jinja
rules:                                all
==== finding fixable violations ====
== [test.sql] FAIL                                                                                               
L:   2 | P:  12 | LT01 | 'data_type' elements are expected to be aligned. Found
                       | incorrect whitespace before 'STRING' keyword: '         
                       | '. [layout.spacing]
L:   2 | P:  33 | LT01 | 'column_constraint_segment' elements are expected to be
                       | aligned. Found incorrect whitespace before 'COMMENT'
                       | keyword: '        '. [layout.spacing]
L:   3 | P:  13 | LT01 | 'data_type' elements are expected to be aligned. Found
                       | incorrect whitespace before 'INT' keyword: '        '.
                       | [layout.spacing]
L:   3 | P:  24 | LT01 | 'column_constraint_segment' elements are expected to be
                       | aligned. Found incorrect whitespace before 'COMMENT'
                       | keyword: '  '. [layout.spacing]
== [test.sql] FIXED
4 fixable linting violations found
```

```bash
cat test.sql                 
CREATE TABLE mytable (
    date_pk  STRING COMMENT 'Date',
    year_num INT    COMMENT 'Year'
)
```

Aka it is fixing as expected.

Also I think it's much better to add comments to the original issue, rather than opening a new with the same problem.

**Comment by Aigul9** [comment]
@peterbud it’s the same as in the previous issue

**Comment by Aigul9** [comment]
@peterbud I found that it happens specifically when I use templated fields
```
CREATE TABLE mytable (
    date_pk               STRING        COMMENT 'Date',
    year_num        INT  COMMENT 'Year',
    ${tech_fields}
) USING ICEBERG
COMMENT 'Справочник систем-источников';
```
where tech_fields = default_column string
